### DISCUSSION TITLE
Request for help with inventories paged-gui click handler

### DISCUSSION THREAD

**Discussion by ShellWen** [NichtStudioCode/InvUI#42] [Q&A] [answered]
I would greatly appreciate some assistance with using PagedGui.inventories() to create a paged GUI, which would allow me to browse Inventories with pagination. However, I'm hoping to kindly inquire if it would be possible to prevent players from taking items from the GUI. I seem to be facing some difficulty in locating where to override the onClick(clickHandler?) method. Any help or guidance you could provide would be sincerely valued.

**Comment by NichtStudioCode** [comment, marked as answer]
Inventories are generally intended to be modified by players. If you want to display items that cannot be modified, you should use InvUI's UI `Item` instead. Nevertheless, you can also cancel inventory events, as explained in the documentation: https://docs.xenondevs.xyz/invui/inventory/#inventory-events

**Reply by ShellWen** [reply]
I am currently facing an issue where the gui does not refresh correspondingly when reading items from the inventory using the Item construct. This has been bothering me for a while.

Thank you very much for always responding to my questions promptly. I apologize for not being able to contribute much to this project, but please allow me to express my gratitude.

**Reply by NichtStudioCode** [reply]
What `Inventory` implementation are you using?

**Reply by ShellWen** [reply]
VirtualInventory

**Reply by NichtStudioCode** [reply]
What exactly do you mean by "items from the inventory using the Item construct"? Are you wrapping VirtualInventory slots in `Item`s? If so, that's not what I meant. I don't know much about your use-case, so I covered all cases. If you really need to use an `Inventory` to display items AND they should not be changed by viewers, I suggest you cancel the inventory event (docs linked above).

**Reply by ShellWen** [reply]
I apologize for not carefully reading the documentation. It seems that `setPreUpdateHandler` is what I need. I will check it out.
